public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -265,6 +266,10 @@ export const DashNav = React.memo<Props>((props) => {
       return [renderTimeControls()];
     }
 
+    if (config.unifiedAlertingEnabled) {
+      buttons.push(<AlertRulesToolbarButton dashboardUid={dashboard.uid} />);
+    }
```

Comment:
he biggest issue is here, we should only show this button if the dashboard has panels with associated alert rules

Reply:
This situation is a bit tricky because we need to wait for an API response before making a decision. So, I added the entire logic to the `AlertRulesToolbarButton` component, which renders `null` instead of a `button` when there are no rules.